src/app/shared/code-editor/code-editor.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, AfterViewInit, Input, ViewChild, ElementRef } from '@angular/core';
+import { ace } from './../grammars/custom-grammars.js';
+
+@Component({
+  selector: 'app-code-editor',
+  templateUrl: './code-editor.component.html',
+  styleUrls: ['./code-editor.component.scss']
+})
+export class CodeEditorComponent implements AfterViewInit {
+  _content: string;
+  editor: any;
+  mode = 'yaml';
+  _filepath: string;
+  aceId: string;
+  @Input() set filepath(filepath: string) {
+    if (filepath !== undefined) {
+      this.setMode(filepath);
+      this._filepath = filepath;
+    }
+  }
+
+  @Input() set content(content: string) {
+    this._content = content;
+    if (this.editor !== undefined && content !== undefined && content !== null) {
+      this.editor.setValue(this._content, -1);
+    }
+  }
+
+  @ViewChild('aceEditor') aceEditor: ElementRef;
+
+  constructor(private elementRef: ElementRef) {
+    this.aceId = Math.floor(Math.random() * 100000).toString();
+  }
+
+  ngAfterViewInit() {
+    const aceMode = 'ace/mode/' + this.mode;
+    this.editor = ace.edit('aceEditor_' + this.aceId,
+      {
+        mode: aceMode,
+        readOnly: true,
+        showLineNumbers: true,
+        maxLines: 60,
+        theme: 'ace/theme/idle_fingers',
+        fontSize: '14pt'
+      }
+    );
+
+    // Set content if possible
+    if (this._content !== null && this._content !== undefined && this._content !== null) {
+      this.editor.setValue(this._content, -1);
+    }
+  }
+
+  /**
+   * Changes the mode of the editor based on the filepath
+   * @param filepath Filepath of file
+   */
+  setMode(filepath: string): void {
+    if (filepath !== undefined && filepath !== null) {
+      if (filepath.endsWith('cwl')) {
+        this.mode = 'cwl';
+      } else if (filepath.endsWith('wdl')) {
+        this.mode = 'wdl';
+      } else if (filepath.includes('Dockerfile')) {
+        this.mode = 'dockerfile';
+      } else if (filepath.endsWith('json')) {
+        this.mode = 'json';
+      } else {
+        this.mode = 'yaml';
```

Comment:
Maybe look for *.yml or *.yaml for 'yaml' 
Otherwise default to text might make more sense 
(Consider CWL workflows which I've seen include JS files for example)

Reply:
I'll make the changes for yaml and text. As for CWL including JS files, you can define sub grammars based on existing grammars. This would be a good future task.

Reply:
I'm actually thinking about workflows that include js files outright 
https://dockstore.org/workflows/github.com/ICGC-TCGA-PanCancer/pcawg-oxog-filter